parserator_web/views.py

```diff
@@ -16,9 +17,42 @@ class AddressParse(APIView):
     def get(self, request):
         # TODO: Flesh out this method to parse an address string using the
         # parse() method and return the parsed components to the frontend.
-        return Response({})
+        address = request.query_params.get('address')
+
+        if not address:
+            return Response({
+                'data': None,
+                'error': {
+                    'message': "Missing 'address' query param "
+                }
+            }, status.HTTP_400_BAD_REQUEST)
+        else:
+            try:
+                address_components, address_type = self.parse(address)
+                return Response({
+                    'data': {
+                        'input_string': address,
+                        'address_components': address_components,
+                        'address_type': address_type
+                    },
+                    'error': None
+                })
+            except Exception as err:
+                return Response({
+                    'data': None,
+                    'error': {
+                        'message': err.__str__()
+                    }
+                }, status.HTTP_500_INTERNAL_SERVER_ERROR)
 
     def parse(self, address):
         # TODO: Implement this method to return the parsed components of a
         # given address using usaddress: https://github.com/datamade/usaddress
-        return address_components, address_type
+
+        try:
```

Comment:
In `try` / `except` blocks, I like to limit the code in the try block to the code that might fail, that way the logic is more clear. It is also safer to handle specific exceptions, rather than all Exceptions, so we can be sure the code is only failing in ways we expect, while unexpected errors are raised.

Can you revise such that:

```
try:
    # code that might fail
except specific exception:
    # code that should execute if the try block fails with this specific exception
# optionally, except Exception:
#    code that should execute if an unexpected error occurs
else:
    # code that should execute if the try block succeeds
```

Reply:
Very helpful. I decided to remove the `try`/`except` from the `parse` method and apply your revision to the route `try`/`except`. Using it in the `parse` method was leading to some redundancy issues. I believe the `RepeatedLabelError` should result in a `400 Bad Request` so let parse raise whatever exception and have the route check for that specific error.

I also print the exception message on the server but return custom error messages to the front end. I was thinking providing messaging more tailored for end users makes more sense then displaying errors meant for developers.